### PR TITLE
[REVIEW] Some clean up

### LINKED ISSUE #387: Clean up engine code

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #387

### CODE REVIEW

**Comment by williamBlazing** [comment]
rerun tests

File: engine/src/cython/engine.cpp [hunk 1]
```diff
@@ -218,18 +216,18 @@ std::unique_ptr<ResultSet> runSkipData(int32_t masterIndex,
...
-		std::vector<std::string> names;
-		for(int i = 0; i < row_groups_cols.size(); i++) {
-			auto& column = row_groups_cols[i];
-			columns.push_back(column.get_gdf_column());
-			// TODO percy cudf0.12 port to cudf::column
-			// GDFRefCounter::getInstance()->deregister_column(column.get_gdf_column());
-
-			names.push_back(column.name());
-		}
+		// auto row_groups_cols = ral::skip_data::process_skipdata_for_table(input_loaders[0], minmax_metadata_tables[0], query, queryContext);
```
Comment (felipeblazing) [inline engine/src/cython/engine.cpp]: porque no se necesita?

File: engine/src/skip_data/SkipDataProcessor.cpp [hunk 2]
```diff
@@ -30,69 +26,69 @@ namespace skip_data {
...
     // return ral::io::create_empty_columns(names, dtypes, time_units, column_indices);
-}
+// }
 
 // "BindableTableScan(table=[[main, customer]], filters=[[OR(AND(<($0, 15000), =($1, 5)), =($0, *($1, $1)), >=($1, 10), <=($2, 500))]], projects=[[0, 3, 5]], aliases=[[c_custkey, c_nationkey, c_acctbal]])"
 //      projects=[[0, 3, 5]]
 // minmax_metadata_table => use these indices [[0, 3, 5]]
 // minmax_metadata_table => minmax_metadata_table[[0, 1,  6, 7,  10, 11, size - 2, size - 1]]
-std::vector<gdf_column_cpp> process_skipdata_for_table(ral::io::data_loader & input_loader, std::vector<gdf_column*> new_minmax_metadata_table, std::string table_scan, const Context& context) {
+// std::vector<gdf_column_cpp> process_skipdata_for_table(ral::io::data_loader & input_loader, std::vector<gdf_column*> new_minmax_metadata_table, std::string table_scan, const Context& context) {
```
Comment (felipeblazing) [inline engine/src/skip_data/SkipDataProcessor.cpp]: ya no se necesita todo esto para skip data?

File: engine/src/skip_data/SkipDataProcessor.h [hunk 3]
```diff
@@ -6,17 +6,16 @@
...
 #include "io/DataLoader.h"
-#include "GDFColumn.cuh"
 
 using blazingdb::manager::experimental::Context;
 
 namespace ral {
 namespace skip_data {
 
-std::vector<gdf_column_cpp> process_skipdata_for_table(ral::io::data_loader & input_loader, std::vector<gdf_column*> new_minmax_metadata_table, std::string table_scan, const Context& context);
+// std::vector<gdf_column_cpp> process_skipdata_for_table(ral::io::data_loader & input_loader, std::vector<gdf_column*> new_minmax_metadata_table, std::string table_scan, const Context& context);
```
Comment (felipeblazing) [inline engine/src/skip_data/SkipDataProcessor.h]: parece que estamos sacando este api?

File: engine/src/cython/engine.cpp (hunk 1 above)
Comment (jeanp413) [reply engine/src/cython/engine.cpp]: just commenting it out to avoid compile errors. I think @williamBlazing is working on skip data migration

**Comment by kharoc** [comment]
1 RAL

*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1135/1210
FAILED: 75/1210
CRASH: 0/1210
TOTAL: 1210
*******************************************************************************
!!!!!!!!!!!!!!!! FAILED gdf TESTS !!!!!!!!!!!!!!!!
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Aggregations without group by Test - gdf
    FAILED: 2/6 ['TEST_05', 'TEST_06']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Coalesce Test - gdf
    FAILED: 3/11 ['TEST_01', 'TEST_04', 'TEST_14']
    CRASH: 0/11
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Count without group by Test - gdf
    FAILED: 1/6 ['TEST_03']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Full outer join Test - gdf
    FAILED: 2/4 ['TEST_01', 'TEST_04']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Left outer join Test - gdf
    FAILED: 2/4 ['TEST_01', 'TEST_02']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Predicates With Nulls Test - gdf
    FAILED: 2/5 ['TEST_02', 'TEST_03']
    CRASH: 0/5
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Tables from Pandas Test - gdf
    FAILED: 1/13 ['TEST_12']
    CRASH: 0/13
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Case Test - gdf
    FAILED: 1/10 ['TEST_13']
    CRASH: 0/10
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Dir Test - gdf
    FAILED: 1/22 ['TEST_12']
    CRASH: 0/22
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Wild Card Test - gdf
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
!!!!!!!!!!!!!!!! FAILED psv TESTS !!!!!!!!!!!!!!!!
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Aggregations without group by Test - psv
    FAILED: 2/6 ['TEST_05', 'TEST_06']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Coalesce Test - psv
    FAILED: 3/11 ['TEST_01', 'TEST_04', 'TEST_14']
    CRASH: 0/11
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Count without group by Test - psv
    FAILED: 1/6 ['TEST_03']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Full outer join Test - psv
    FAILED: 2/4 ['TEST_01', 'TEST_04']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Left outer join Test - psv
    FAILED: 2/4 ['TEST_01', 'TEST_02']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Predicates With Nulls Test - psv
    FAILED: 2/5 ['TEST_02', 'TEST_03']
    CRASH: 0/5
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Case Test - psv
    FAILED: 1/10 ['TEST_13']
    CRASH: 0/10
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: File System Local Test - psv
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: File System S3 Test - psv
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Wild Card Test - psv
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
!!!!!!!!!!!!!!!! FAILED parquet TESTS !!!!!!!!!!!!
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Aggregations without group by Test - parquet
    FAILED: 2/6 ['TEST_05', 'TEST_06']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Coalesce Test - parquet
    FAILED: 3/11 ['TEST_01', 'TEST_04', 'TEST_14']
    CRASH: 0/11
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Count without group by Test - parquet
    FAILED: 1/6 ['TEST_03']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Full outer join Test - parquet
    FAILED: 2/4 ['TEST_01', 'TEST_04']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Left outer join Test - parquet
    FAILED: 2/4 ['TEST_01', 'TEST_02']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Predicates With Nulls Test - parquet
    FAILED: 2/5 ['TEST_02', 'TEST_03']
    CRASH: 0/5
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Case Test - parquet
    FAILED: 1/10 ['TEST_13']
    CRASH: 0/10
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: File System Local Test - parquet
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: File System S3 Test - parquet
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Wild Card Test - parquet
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
!!!!!!!!!!!!!!!! FAILED orc TESTS !!!!!!!!!!!!!!!!
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Aggregations without group by Test - orc
    FAILED: 2/6 ['TEST_05', 'TEST_06']
    CRASH: 0/6
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Coalesce Test - orc
    FAILED: 3/11 ['TEST_01', 'TEST_04', 'TEST_14']
    CRASH: 0/11
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Date Test - orc
    FAILED: 1/5 ['TEST_01']
    CRASH: 0/5
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Timestamp Test - orc
    FAILED: 1/12 ['TEST_02']
    CRASH: 0/12
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Full outer join Test - orc
    FAILED: 2/4 ['TEST_01', 'TEST_04']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Left outer join Test - orc
    FAILED: 2/4 ['TEST_01', 'TEST_02']
    CRASH: 0/4
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Predicates With Nulls Test - orc
    FAILED: 2/5 ['TEST_02', 'TEST_03']
    CRASH: 0/5
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Bindable Alias Test - orc
    FAILED: 1/12 ['TEST_09']
    CRASH: 0/12
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: File System Local Test - orc
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Simple Distribution From Local Test - orc
    FAILED: 4/52 ['TEST_22', 'TEST_43', 'TEST_46', 'TEST_49']
    CRASH: 0/52
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: Wild Card Test - orc
    FAILED: 1/21 ['TEST_12']
    CRASH: 0/21
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    SUMMARY for FAILED test suite: TPCH Queries Test - orc
    FAILED: 6/11 ['TEST_03', 'TEST_05', 'TEST_06', 'TEST_10', 'TEST_14', 'TEST_15', 'TEST_18']
    CRASH: 0/11
!!!!!!!!!!!!!!!! FAILED dask_cudf TESTS !!!!!!!!!!!!!!!!
MAX DELTA: 36.0
*******************************************************************************
Aggregations without group by Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Coalesce Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Column Basis Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Common Table Expressions Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Count without group by Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Date Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Timestamp Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Full outer join Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Group by Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Group by without aggregations Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Inner join Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Left outer join Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Non-EquiJoin Queries Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Nested Queries Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Order by Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Predicates With Nulls Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Simple String Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Unary ops Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Inner join Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Union Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Limit Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Where clause Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Bindable Alias Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Boolean Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Case Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Cast Test:   Start Mem: 3455.1875   End Mem: 3455.1875   Diff: 0.0
Concat Test:   Start Mem: 3455.1875   End Mem: 3491.1875   Diff: 36.0
Literal Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
Dir Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
File System Local Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
File System S3 Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
Like Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
Substring Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
Wild Card Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0
TPCH Queries Test:   Start Mem: 3491.1875   End Mem: 3491.1875   Diff: 0.0

2 Workers


Test | Resultados
-- | --
aggregationsWithoutGroupByTest | FAILED: 10/30 ['TEST_05', 'TEST_06']
coalesceTest | CRASH
columnBasisTest |  
commonTableExpressionsTest | OK
countWithoutGroupByTest | FAILED: 4/24 ['TEST_03']
dateTest | CRASH
timestampTest | CRASH (memory)
fullOuterJoinsTest | CRASH
groupByTest | OK
GroupByWitoutAggregations | OK
innerJoinsTest | OK
leftOuterJoinsTest | FAILED: 10/20  ['TEST_01', 'TEST_02']
nonEquiJoinsTest | CRASH
nestedQueriesTest | OK
orderbyTest | OK
predicatesWithNulls | CRASH
stringTests | CRASH
tablesFromPandasTest | CRASH
unaryOpsTest | OK
unifyTablesTest | OK
unionTest | OK
useLimitTest | CRASH
whereClauseTest | OK
bindableAliasTest | CRASH
booleanTest | CRASH
caseTest | CRASH
castTest | FAILED: 3/39 ['TEST_01']
concatTest | FAILED: 4/5
literalTest | CRASH
dirTest | CRASH
fileSystemLocalTest | CRASH
likeTest | CRASH
simpleDistributionTest | CRASH
substringTest | CRASH
wildCardTest | CRASH
tpchQueriesTest | CRASH

**Pull request merged**